packages/ethereum/Dockerfile.hardhat

```diff
@@ -0,0 +1,57 @@
+ARG HOPR_TOOLCHAIN_IMAGE=${HOPR_TOOLCHAIN_IMAGE:-hopr-toolchain:latest}
+
+# Creates a Docker container that encapsulates a local testnet with deployed smart contracts
+
+FROM ${HOPR_TOOLCHAIN_IMAGE} as build
+
+WORKDIR /app/hoprnet
+
+# Get all we need from the monorepo
+COPY .yarn .yarn/
+COPY packages packages/
+COPY package.json .yarnrc.yml yarn.lock tsconfig.json Makefile Cargo.toml Cargo.lock ./
+
+LABEL description="Launches a hardhat node running a local network with the HOPR contracts deployed and available." 
+
+# Turn Ethereum package into hardhat package
+RUN cd /app/hoprnet/packages/ethereum && \
+    mv package.hardhat.json package.json && \
+    mv tsconfig.hardhat.json tsconfig.json && \
+    cd /app/hoprnet
+
+RUN CI=true make deps package=hopr-ethereum
+
+RUN NO_NEXT=true make build package=ethereum
+
+# Remove all unused packages
+RUN rm -R packages/avado packages/connect packages/cover-traffic-daemon packages/core-ethereum packages/hoprd
+
+WORKDIR /app/hoprnet/packages/ethereum/
+
+# # use alpine version of node for smallest image sizes
+FROM node:16-alpine as runtime
+
+WORKDIR /app/hardhat
+
+RUN apk add --no-cache tini libc6-compat make
```

Comment:
Although it might difficult reading, on runtime images we should take care of disk image size, then I would encourage to join `COPY` commands one next to the other and join all the `RUN` commands in just one RUN command appending them with &&. The same comment apply to other Dockerfiles

Reply:
changed, now one long command